Tests/NIOTests/ChannelTests.swift

```diff
@@ -1526,4 +1526,89 @@ public class ChannelTests: XCTestCase {
         // The read should go through.
         XCTAssertNoThrow(try readFuture.wait())
     }
+
+    func testNoChannelReadIfNoAutoRead() throws {
+        let group = MultiThreadedEventLoopGroup(numThreads: 1)
+        defer {
+            XCTAssertNoThrow(try group.syncShutdownGracefully())
+        }
+
+        class NoChannelReadVerificationHandler: ChannelInboundHandler {
+            typealias InboundIn = ByteBuffer
+
+            public func channelRead(ctx: ChannelHandlerContext, data: NIOAny) {
+                XCTFail("Should not be called as autoRead is false and we did not call read(), but received \(self.unwrapInboundIn(data))")
+            }
+        }
+
+        let serverChannel = try ServerBootstrap(group: group)
+            .serverChannelOption(ChannelOptions.socket(SocketOptionLevel(SOL_SOCKET), SO_REUSEADDR), value: 1)
+            .childChannelOption(ChannelOptions.autoRead, value: false)
+            .childChannelInitializer { ch in
+                ch.pipeline.add(handler: NoChannelReadVerificationHandler())
+            }
+            .bind(host: "127.0.0.1", port: 0).wait()
+
+        let clientChannel = try ClientBootstrap(group: group)
+            .connect(to: serverChannel.localAddress!).wait()
+        var buffer = clientChannel.allocator.buffer(capacity: 8)
+        buffer.write(string: "test")
+        try clientChannel.writeAndFlush(buffer).wait()
```

Comment:
I think we need a wait here, right?

Reply:
there is a `wait`... what I am missing ? Or you talk about waiting for some time and verify that channelRead was not triggered ?

Reply:
The second one. :smile:

Reply:
done